### PR TITLE
Issue9 mmd validator

### PR DESCRIPTION
**Pull request by mortenwh**
**Summary**: xml validation

**Related issue**: #9

**Suggested reviewer(s)**:

**Reviewer checklist**:

* [ ] The headers of all files contain a reference to the repository license
* [ ] 100% test coverage of new code - meaning:
  * [ ] The overall test coverage increased or remained the same as before
  * [ ] Every function is accompanied with a test suite
  * [ ] Tests are both positive (testing that the function work as intended with valid data) and negative (testing that the function behaves as expected with invalid data, e.g., that correct exceptions are thrown)
  * [ ] Functions with optional arguments have separate tests for all options
* [ ] Examples are supported by doctests
* [ ] All tests are passing
* [ ] All names (e.g., files, classes, functions, variables) are explicit
* [ ] Documentation (as docstrings) is complete and understandable

The checklist is based on the S-ENDA conventions and definition of done (see [General Conventions](https://s-enda-documentation.readthedocs.io/en/latest/general_conventions.html)). The above points are not necessarily relevant to all contributions. In that case, please add a short explanation to help the reviewer.

### CODE REVIEW

**Comment by vkbo** [comment]
I'm making a new PR where we extract what we need from py_mmd_tools into a folder called `external/py_mmd_tools`. This is the solution that for now requires the least amount of work until the `py_mmd_tools` repo can be fixed.

The changes made here could instead be added on that PR.

File: tests/test_api/test_worker.py [hunk 1]
```diff
@@ -87,4 +80,35 @@ def testApiWorker_Distributor(tmpDir, tmpConf, monkeypatch):
...
+def testCoreWorker_validator_fails(monkeypatch):
+    """Test the Worker class validator.
+    """
+    current_folder = os.path.dirname(__file__)
+    fn = os.path.join(current_folder,
+            '../files/api/aqua-modis-1km-20210429002844-20210429003955.xml')
+    data = readFile(fn)
+    def mock_check_information_content(*args, **kwargs):
+        return True, ""
+    monkeypatch.setattr(Worker, '_check_information_content', mock_check_information_content)
```
Comment (vkbo) [inline tests/test_api/test_worker.py:96]: I would prefer it if we could keep one test per function, and instead split up mocking by using the monkeypatch context facility to test different scenarios. It makes the code and test output a lot less verbose.

File: dmci/config.py [hunk 2]
```diff
@@ -37,6 +37,8 @@ def __init__(self):
 
         # API
         self.distributor_input_path = None
+        self.mmd_xsd_schema = os.path.join(self.pkgRoot, 'dmci/assets/mmd.xsd')
```
[the remark this answers is not in the document]
Comment (mortenwh) [reply dmci/config.py:40]: done

File: tests/test_api/test_worker.py (hunk 1 above)
Comment (mortenwh) [reply tests/test_api/test_worker.py:96]: ok - I changed the code. Is it better?

**Comment by mortenwh** [comment]
Nå er skal testene være fullstendig, men den på api'et feiler fortsatt.. Og hva gjør vi med externals-pakken? Skal vi bare ignorere den?

File: tests/test_api/test_worker.py (hunk 1 above)
Comment (vkbo) [reply tests/test_api/test_worker.py]: I can't see any change in the code here

Comment (mortenwh) [reply tests/test_api/test_worker.py]: I'm not sure if I understand - one test per function meaning one test-function per function that should be tested, or one test-function per test (i.e., per assertion)? Also, context facility and monkeypatching is new to me, and I didn't take time for reading up... It does the job, doesn't it? ;-)

**Comment by vkbo** [comment]
I made a few minor changes. I fixed the `os.path.join()` calls, and removed some code from the new tests that there is already a fixture for.

That one API test still fails because a return code changed, but I think @johtoblan should have a look at that.

**Review by vkbo — APPROVED** [review]
Looks good!

**Pull request merged**